src/index.js

```diff
@@ -86,6 +86,12 @@ function fetch (uri, opts) {
           reject(new FetchError(`redirect location header missing at: ${request.url}`, 'invalid-redirect'))
           return
         }
+        // Remove authorization if changing hostnames (but not if just
+        // changing ports or protocols).  This matches the behavior of request:
+        // https://github.com/request/request/blob/b12a6245/lib/redirect.js#L134-L138
+        if (url.parse(request.url).hostname !== url.parse(res.headers.location).hostname) {
```

Comment:
Correction again: It allows _protocol-relative URIs_ (that is, `//foo.com/bar`). So that still needs to be compensated for. Heh.